instrumentation.go

```diff
@@ -243,7 +244,11 @@ func (c instConfig) res() *resource.Resource {
 		semconv.SchemaURL,
 		semconv.ServiceNameKey.String(c.serviceName),
 		semconv.TelemetrySDKLanguageGo,
+		// will be removed from semconv in the future
 		semconv.TelemetryAutoVersionKey.String(Version()),
+		// will be added to semconv in the future
```

Comment:
Wouldn't it be better to do it after semconv bump (to v1.21)??

Reply:
I checked and the [v1.21](https://pkg.go.dev/go.opentelemetry.io/otel@v1.21.0/semconv/v1.21.0) package didn't include the new keys. 

<img width="466" alt="image" src="https://github.com/open-telemetry/opentelemetry-go-instrumentation/assets/3481731/8d0f98cf-e380-4eba-a352-facbb7bd962b">

Reply:
We could wait for new semconv package. I don't see the harm in adding early as we'll want to duplicate the fields for a period of time anyway.

Reply:
I meant 1.23 :facepalm: